routes/helpers/postgres.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+class Clover
+  def postgres_post(name)
+    Authorization.authorize(current_account.id, "Postgres:create", @project.id)
+    fail Validation::ValidationFailed.new({billing_info: "Project doesn't have valid billing information"}) unless @project.has_valid_payment_method?
+
+    Validation.validate_postgres_location(@location)
+
+    required_parameters = ["size"]
+    required_parameters << "name" << "location" if web?
+    allowed_optional_parameters = ["storage_size", "ha_type", "version", "flavor"]
+    request_body_params = Validation.validate_request_body(json_params, required_parameters, allowed_optional_parameters)
+    parsed_size = Validation.validate_postgres_size(@location, request_body_params["size"])
+
+    ha_type = request_body_params["ha_type"] || PostgresResource::HaType::NONE
+    requested_standby_count = case ha_type
+    when PostgresResource::HaType::ASYNC then 1
+    when PostgresResource::HaType::SYNC then 2
+    else 0
+    end
+
+    requested_postgres_core_count = (requested_standby_count + 1) * parsed_size.vcpu / 2
+    Validation.validate_core_quota(@project, "PostgresCores", requested_postgres_core_count)
+
+    st = Prog::Postgres::PostgresResourceNexus.assemble(
+      project_id: @project.id,
+      location: @location,
+      name:,
+      target_vm_size: parsed_size.vm_size,
+      target_storage_size_gib: request_body_params["storage_size"] || parsed_size.storage_size_options.first,
+      ha_type: request_body_params["ha_type"] || PostgresResource::HaType::NONE,
+      version: request_body_params["version"] || PostgresResource::DEFAULT_VERSION,
+      flavor: request_body_params["flavor"] || PostgresResource::Flavor::STANDARD
+    )
+    send_notification_mail_to_partners(st.subject, current_account.email)
+
+    if api?
+      Serializers::Postgres.serialize(st.subject, {detailed: true})
+    else
+      flash["notice"] = "'#{name}' will be ready in a few minutes"
+      request.redirect "#{@project.path}#{PostgresResource[st.id].path}"
```

Comment:
Is the PostgresResource retrievable from `st` without issuing an additional query? (I'm not sure)

Reply:
I don't know a way to do it without an extra query, but we can access it through `st.subject.id`

Reply:
I think it is not without changing the nexus to return the `PostgresResource` instead of `Strand`, which would have broader implications as many places expect it to return `Strand`. I can change this to `st.subject.path` if you think it is more readable, but it wouldn't help with the query count.